lib/deploy.rb

```diff
@@ -22,24 +24,42 @@ def run
       check_for_unstaged_changes!
       check_for_changelog!
       set_aws_region!
+      fetch_eb
       verify_configuration!
 
-      @name, @app_bucket = select_app
-      @platform          = detect_platform
+      @name       = deployment_target
+      @platform   = detect_platform
 
       request_confirmation!
       synchronize_repo!
       deploy!
```

Comment:
Well, I still see an object whose initialisation triggers actions instead of setting initial state. It's about clarity and maintainability.